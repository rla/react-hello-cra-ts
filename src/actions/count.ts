export const SET_COUNT = 'SET_COUNT';
export const INCREASE_COUNT = 'INCREASE_COUNT';

export type SetCountAction = {
  type: typeof SET_COUNT;
  value: number;
};

export type IncreaseCountAction = {
  type: typeof INCREASE_COUNT;
};

export type CountAction = SetCountAction | IncreaseCountAction;

export const setCount = (value: number): SetCountAction =>
  ({ type: SET_COUNT, value });

export const increaseCount = (): IncreaseCountAction =>
  ({ type: INCREASE_COUNT });
