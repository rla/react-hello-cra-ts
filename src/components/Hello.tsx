import { CSSProperties } from 'react';

type Props = {
  name: string;
};

export const Hello = ({ name }: Props) => {
  const style: CSSProperties = {
    color: 'red'
  };
  return <h1 style={style}>Hello {name}</h1>;
};