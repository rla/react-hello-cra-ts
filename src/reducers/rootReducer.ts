import { combineReducers } from 'redux'
import { authReducer, AuthState } from './authReducer';
import { countReducer, CountState } from './countReducer'
import { profileReducer, ProfileState } from './profileReducer';

export type RootState = {
  auth: AuthState;
  count: CountState;
  profile: ProfileState;
};

export const rootReducer = combineReducers<RootState>({
  auth: authReducer,
  count: countReducer,
  profile: profileReducer
});
