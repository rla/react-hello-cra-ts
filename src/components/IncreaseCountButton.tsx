import { useDispatch } from 'react-redux';
import { increaseCount } from '../actions/count';

export const IncreaseCountButton = () => {
  const dispatch = useDispatch();
  const handleClick = () => dispatch(increaseCount());
  return (
    <button data-testid="plusone-button" type="button" onClick={handleClick}>
      +1
    </button>
  );
};
