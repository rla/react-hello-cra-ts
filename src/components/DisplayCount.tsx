import { useSelector } from 'react-redux';
import { RootState } from '../reducers/rootReducer';

export const DisplayCount = () => {
  const count = useSelector((state: RootState) => state.count.count);
  return <span data-testid="display-count">Vajutatud: {count}</span>;
};
