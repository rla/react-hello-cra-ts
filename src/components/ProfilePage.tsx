import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { setProfileData } from '../actions/profile';
import { ProfileData } from '../ProfileData';
import { RootState } from '../reducers/rootReducer';

export const ProfilePage = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    (async () => {
      const response = await fetch('http://localhost:4040/profile');
      const json = await response.json();
      dispatch(setProfileData(json as ProfileData));
    })();
  }, [dispatch]);
  const profileData = useSelector((state: RootState) => state.profile.data);
  if (profileData) {
    return (
      <div>
        <div>
          Nimi: {profileData.name}
        </div>
        <div>
          Email: {profileData.email}
        </div>
        <div>
          <img src={profileData.photo} alt="Profile" width={150} height={150} />
        </div>
      </div>
    );
  } else {
    return <div>Laen andmeid ...</div>
  }
};
