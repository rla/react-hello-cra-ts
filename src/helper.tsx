import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { rootReducer } from './reducers/rootReducer';

export const snapshotTest = (Component: any) => {
  const store = createStore(rootReducer);
  expect((mount(
    <Provider store={store}>
      <Component />
    </Provider>
  ).html())).toMatchSnapshot();
};
