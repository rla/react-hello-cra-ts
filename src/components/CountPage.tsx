import { DisplayCount } from './DisplayCount';
import { IncreaseCountButton } from './IncreaseCountButton';

export const CountPage = () => {
  return (
    <div style={{padding: '100px'}}>
      <div>
        <IncreaseCountButton />
      </div>
      <div>
        <DisplayCount />
      </div>
    </div>
  );
};
