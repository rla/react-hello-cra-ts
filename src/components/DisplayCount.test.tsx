import React from 'react';
import { render, screen } from '@testing-library/react';
import { createStore } from 'redux';
import { rootReducer } from '../reducers/rootReducer';
import { Provider } from 'react-redux';
import { DisplayCount } from '../components/DisplayCount';
import { increaseCount } from '../actions/count';
import { snapshotTest } from '../helper';

test('renders DisplayCount component', () => {
  const store = createStore(rootReducer);
  store.dispatch(increaseCount());
  render(
    <Provider store={store}>
      <DisplayCount />
    </Provider>
  );
  const spanElement = screen.getByTestId('display-count');
  expect(spanElement).toHaveTextContent('Vajutatud: 1');
});

test('snapshot', () => {
  snapshotTest(DisplayCount);
});
