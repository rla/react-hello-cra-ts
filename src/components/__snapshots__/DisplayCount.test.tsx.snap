// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `"<span data-testid=\\"display-count\\">Vajutatud: 0</span>"`;
